### PR TITLE
change linting error messages for null values in arrays

### LINKED ISSUE #11627: Linting error on null in dependencies is misleading
When doing `helm lint` on a chart with the following dependencies: 

```
dependencies:
  - name: foo
    version: 1.2.3
  - null
```

it fails with 
```
==> Linting .
[ERROR] templates/: validation: dependency cannot be an empty list
```

This is quite misleading, as a chart with an empty list (`dependencies: []`) does not fail linting.

This was caused by #11044 and I would suggest to simply change the error message to a more meaningful one. 

Output of `helm version`: 
```
version.BuildInfo{Version:"v3.10.2", GitCommit:"50f003e5ee8704ec937a756c646870227d7c8b58", GitTreeState:"clean", GoVersion:"go1.19.3"}
```

### PR DESCRIPTION
**Pull request by dastrobu**
Closes #11627

Signed-off-by: Daniel Strobusch <1847260+dastrobu@users.noreply.github.com>

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

### CODE REVIEW

File: pkg/chart/metadata_test.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
 
-    http://www.apache.org/licenses/LICENSE-2.0
```
Comment (hickeyma) [inline pkg/chart/metadata_test.go:8]: remove indentation

**Review by hickeyma — CHANGES_REQUESTED** [review]
Looks good bar a small nit

File: pkg/chart/metadata_test.go (hunk 1 above)
Comment (dastrobu) [reply pkg/chart/metadata_test.go]: @hickeyma I fixed this for this PR. 

It is a bit strange, though, to have part of the indentation with spaces and part with tabs, which is why my IDE changed this automatically: 

image

I noticed that this seems to be something that is not consistent over the codebase. Not sure, if there is some discussion ongoing about this issue somewhere....

**Review by hickeyma — APPROVED** [review]

File: pkg/chart/metadata_test.go (hunk 1 above)
Comment (hickeyma) [reply pkg/chart/metadata_test.go]: Not sure if this was ever raised. If you want, you could raise a issue and some examples in it?

**Pull request merged**